src/js/app.js

```diff
@@ -547,10 +549,20 @@
             for (let rule in rules) {
                 let ruleValue = rules[rule];
 
-                if (rule !== RULE_REQUIRED && value == '') {
+                if (rule !== RULE_REQUIRED && rule !== RULE_FUNCTION && value == '') {
                     return;
                 }
                 switch (rule) {
+                    case RULE_FUNCTION:{
+                        if(!ruleValue){
+                            break;
+                        }
+                        if(ruleValue(name, value)){
+                            break;
+                        }
+                        this.generateMessage(RULE_FUNCTION, name, ruleValue);
+                        return;
+                    }
```

Comment:
Pls, small change formatting too. And check for typeof function (just in case).

```
if (!ruleValue) {
                            break;
                        }

                        if (typeof ruleValue !== 'function') {
                            console.warn('Function rule must be a function');
                            break;
                        }

                        if (ruleValue(name, value)) {
                            break;
                        }
```